src/HubSession.ts

```diff
@@ -45,35 +45,48 @@ export interface HubSessionOptions {
    * available in-memory.
    */
   cryptoSuites?: CryptoSuite[];
+
+  /**
+   * Instance of KeyStore than can be used
+   * to get and save keys.
+   */
+  keyStore: IKeyStore;
 }
 
 /**
  * Represents a communication session with a particular Hub instance.
  */
 export default class HubSession {
 
-  private authentication: Authentication;
   private clientDid: string;
-  private clientPrivateKey: PrivateKey;
   private hubDid: string;
   private hubEndpoint: string;
   private targetDid: string;
-
+  private resolver: IDidResolver;
```

Comment:
Could possibly be removed if `resolver` is only used in the constructor?

Reply:
Same for `keyStore` and `clientPrivateKeyReference`

Reply:
removed